py/gc.c

```diff
@@ -152,14 +146,19 @@ void gc_init(void *start, void *end) {
     memset(MP_STATE_MEM(gc_finaliser_table_start), 0, gc_finaliser_table_byte_len);
 #endif
 
-    // set last free ATB index to start of heap
+    // Set first free ATB index to the start of the heap.
     MP_STATE_MEM(gc_last_free_atb_index) = 0;
+    // Set last free ATB index to the end of the heap.
+    MP_STATE_MEM(gc_last_free_atb_index) = MP_STATE_MEM(gc_alloc_table_byte_len) - 1;
```

Comment:
line 150 and 152 are both setting `MP_STATE_MEM(gc_last_free_atb_index)`, so line 150 is wrong or redundant?

Reply:
150 was wrong. Good catch!